llm/ggml.go

```diff
@@ -81,6 +84,50 @@ func (kv KV) ContextLength() uint64 {
 	return kv.u64(fmt.Sprintf("%s.context_length", kv.Architecture()))
 }
 
+func (kv KV) ChatTemplate() string {
+	if tmpl, ok := kv["tokenizer.chat_template"].(string); ok {
+		switch {
+		case strings.Contains(tmpl, "<|im_start|>"):
+			return "chatml"
+		case strings.Contains(tmpl, "<<SYS>>"):
+			// llama2-chat must appear before mistral
+			return "llama2-chat"
+		case strings.Contains(tmpl, "<|start_header_id|>"):
+			return "llama3-instruct"
+		case strings.Contains(tmpl, "[INST]"):
+			return "mistral-instruct"
+		case strings.Contains(tmpl, "<start_of_turn>"):
+			return "gemma-instruct"
+		case strings.Contains(tmpl, "<|assistant|>"):
```

Comment:
I think zephyr might use this these tags too, but haven't checked if it's ultimately the same as Phi

Reply:
these two looks very similar however they're have different ending markers. phi-3 uses `<|end|>` while zephyr uses `</s>`